soundcard/pulseaudio.py

```diff
@@ -86,6 +86,33 @@ def _block_operation(self, operation):
         while self._pa_operation_get_state(operation) == _pa.PA_OPERATION_RUNNING:
             time.sleep(0.001)
 
+    @property
+    def name(self):
+        """Return application name stored in client proplist"""
+        idx = self._pa_context_get_index(self.context)
+        # TODO: idx can be PA_INVALID_INDEX now
+        name = None
+        @_ffi.callback("pa_client_info_cb_t")
+        def callback(context, client_info, eol, userdata):
+            nonlocal name
+            if not eol:
+                name = _ffi.string(client_info.name).decode('utf-8')
+        self._pa_context_get_client_info(self.context, idx, callback, _ffi.NULL)
+        assert name is not None
```

Comment:
What should happen if `eol` is truthy? Exception? Or is this a programming error and an AssertionError is acceptable?

Reply:
That depends on pulse's API, I would say.

Does pulse guarantee/enforce a valid UTF-8 name? If so, the assertion is fine by me. If not, returning an empty string is probably preferable. (Can you set the name to empty string? What does it return in that case?)

Reply:
> Does pulse guarantee/enforce a valid UTF-8 name?

yes.

> Can you set the name to empty string? What does it return in that case?

yes, and it returns an empty string.